### PR TITLE
Improve and complement figures

### LINKED ISSUE #15: Further improve and complement figures
- Address open comments in #14
- Include CMIP5 figures if available
- Add placeholder for unavailable figures or make it impossible to select them
- ...

### PR DESCRIPTION
**Pull request by geek-yang**
- Address open comments in #14
- Include CMIP5 figures if available
- Add placeholder for unavailable figures or make it impossible to select them

close #15

### CODE REVIEW

File: python/maps_prototype_tas.ipynb [hunk 1]
```diff
@@ -136,17 +137,17 @@
     "ax.set_title('EUR temperature difference (%) 50prec', fontsize=20)\n",
```
Comment (Peter9192) [inline python/maps_prototype_tas.ipynb:137]: Unit for temperature is degrees C or K, not %. Also typo in "50prec"

**Review by Peter9192 — CHANGES_REQUESTED** [review]
Very nice work on improving the figures. One major comment: I see no more difference in precip maps for ClimWIP. Could it be that there's a small error there somewhere? Comparing the cleanup notebooks, they seem to be a few orders of magnitude smaller than REA. Perhaps the original units were different? Interestingly, in your precip example plot notebook, the values seem much higher.

Future work: the placeholder says "current selection of parameters". It might be nice to state the parameters explicitly. Perhaps we could use SVG to dynamically generate that placeholder image.

File: python/maps_prototype_tas.ipynb (hunk 1 above)
Comment (geek-yang) [reply python/maps_prototype_tas.ipynb:137]: Corrected!

File: python/maps_creator_atlas_data.py [hunk 2]
```diff
@@ -91,19 +105,19 @@ def plot(data, lat, lon, variable, project, model,
...
+        ax.set_title("\n".join(wrap(f'{method} {cons[constrained]} {season.lower()} relative precipitation projections (%) - {percentile} percentile projected changes for 2050 with respect to present-day climate', 60)), fontsize=20)
     elif variable == "tas":
-        ax.set_title(f'EUR temperature difference (K) {percentile}perc', fontsize=20)
-    plt.show()
-    fig.savefig(os.path.join(output_path,
-                f"eur_{model}_{method}_{variable}_41-60_{season.lower()}_{project.lower()}_{percentile}perc_{cons[constrained]}.png"),
+        ax.set_title("\n".join(wrap(f'{method} {cons[constrained]} {season.lower()} temperature projections (degC) - {percentile} percentile projected changes for 2050 with respect to present-day climate', 60)), fontsize=20)
+    #plt.show()
+    fig.savefig(Path(output_path,
+                f"eur_{method}_{variable}_41-60_{season.lower()}_{project.lower()}_{percentile}perc_{cons[constrained]}.png"),
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply python/maps_creator_atlas_data.py:120]: I got pep8, but just noticed it was not configured correctly in my vs code. Now it is working! Thanks for noticing it.

**Comment by geek-yang** [comment]
Regarding the precip maps for ClimWIP, if seems we use the wrong data for plotting. The ETHZ team provides one type of file for tas (e.g. `eur_tas_41-60_djf_cmip6_10perc.nc`), but two types for pre (e.g. `eur_pr_41-60_djf_cmip6_10perc.nc` and `eur_pr_41-60_djf_cmip6_10perc_rel.nc`). We used files with the same naming convention for both the tas and pre. But in terms of the values, it looks like `eur_pr_41-60_djf_cmip6_10perc_rel.nc` files are the right options. I will update the maps using the corrected data. Thanks a lot for finding this @Peter9192 ! ^ ^

File: python/README.md [hunk 3]
```diff
@@ -5,14 +5,26 @@ The scripts and notebooks contained in this folder are used to generate content
...
 
+The model outputs are preprocessed and saved as netcdf4 files, following the format:
+**coordinates:**
+- lon (lon) (float64) [-10, ..., 40]
+- lat (lat) (float64) [30, ..., 75]
+- percentile (percentile) (int64) [10, 25, 50, 75, 90]
+- constrained (constrained) (int64) [1, 0]
+- season (season) (object) ['DJF', 'JJA']
+
+**variables**
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply python/README.md:16]: Sounds good! We can think about it and possibly implement it in our next pull request.

File: python/maps_creator_atlas_data.py [hunk 4]
```diff
@@ -60,17 +68,23 @@ def prepareData(nc_file, datapath, output_path):
...
-                     model, method, s, c, p, output_path)
-                # plot precipitation
-                plot(data_pr, lat, lon, "pr", project,
-                     model, method, s, c, p, output_path)
+                try:
+                    data_pr = dataset['pr'][key_s[s],key_c[c],key_p[p],:,:]
+                    # plot precipitation
+                    plot(data_pr, lat, lon, "pr", project,
+                         method, s, c, p, output_path)
+                except:
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply python/maps_creator_atlas_data.py:78]: Thanks for your comment @Peter9192. I agree that it is good practice to be specific about the exception. It is `IndexError` in this case and it is now added to the code.

File: python/maps_creator_atlas_data.py [hunk 5]
```diff
@@ -60,17 +68,23 @@ def prepareData(nc_file, datapath, output_path):
     for s in season:
         for c in constrained:
             for p in percentile:
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply python/maps_creator_atlas_data.py:72]: Hi Peter, thanks for the refactoring. I follow your suggestion in # 21 and use `enumerate`. Looks much clean now!

**Pull request merged**